src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -360,7 +360,11 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
 
   /* End of size map handling code */
 
-  protected def elemEquals(key1: A, key2: A): Boolean = (key1 == key2)
+  protected def elemEquals(key1: A, key2: A): Boolean = {
+    if (key1.asInstanceOf[AnyRef] eq key2.asInstanceOf[AnyRef]) true
+    else if (key1.getClass eq key2.getClass) key1.equals(key2)
```

Comment:
Needs better NPE protection here if only one of `key1` or `key2` is null.

Reply:
This test is expensive in Scala.js, not an optimization at all. Could the contents of this method be factored somewhere where we can more easily override it in Scala.js, instead of having to copy-paste the whole `HashMap.scala` file?